### DISCUSSION TITLE
Add thumbnail images to the replicator sets?

### DISCUSSION THREAD

**Discussion by bobevsky** [statamic/cms#10356] [Polls]
Hey everyone,

I’ve been thinking about the user experience with the replicator sets. Many of us are currently using the replicator to build pages with blocks (widgets), and when there are a lot of blocks available, it can be confusing to know which block to use. In Statamic 4, you added the option to add icons to the blocks, but should we include thumbnail images for the sets as well? Many of our clients have requested this as well (because we had it in the previous CMS before switching to Statamic).

<img width="766" alt="image" src="https://github.com/statamic/cms/assets/44194428/855cad32-d1c8-4bb5-8616-d95e8e60805f">

Here are a few points to consider:

1. Improved User Experience: Thumbnails could make it easier for users to quickly identify the sets they need, especially if they are visual learners.
2. Enhanced Visual Appeal: Adding images could make the interface look more modern and attractive. (I am sure some UI changes to the sets picker will be needed)
3. Clarity and Precision: Sometimes, text descriptions aren't enough, and an image can prevent misunderstandings about what the set includes.

**Comment by jackmcdade** [comment]
You can point your Replicator sets at a different directory to load your own images already :) 
https://statamic.dev/fieldtypes/replicator#custom-set-icons

**Reply by bobevsky** [reply]
I am aware of this, but still, it's an icon (it must be svg? would be great if we can just upload screenshots of the components in png), and is really small. Maybe you can add the functionality so when you hover over it, it enlarges into a bigger image so you can clearly see what this block is about? It's just an idea, or maybe re-do the entire UI of the sets picker to support bigger images?

<img width="792" alt="image" src="https://github.com/statamic/cms/assets/44194428/5545e5a8-a1af-4862-89e1-4d21b3548856">

**Reply by bobevsky** [reply]
Just so you get the idea, this is what i am aiming for:

![image](https://github.com/statamic/cms/assets/44194428/1ab88237-2b1b-47e0-a260-049da1f568a6)

**Reply by edalzell** [reply]
Feel free to add a feature request on the `ideas` repo: https://github.com/statamic/ideas

**Reply by vitalijalbu** [reply]
any updates about this guys??

**Reply by edalzell** [reply]
You'll get notified if/when the idea is implemented.

**Reply by theLeroy** [reply]
i would love to see this implemented

**Reply by jackmcdade** [reply]
It's in v6.
